docs/QUICKSTART.md

```diff
@@ -0,0 +1,410 @@
+
+```C
+/* Number of timer samples we will use */
+#define PRIVATE_TIMER_BUFFER_COUNT 256
+
+/* For debug purposes: Print the buffer values when a button is pressed. */
+static void private_timer_button_on_press(void)
+{
+    for (int i = 0; i < PRIVATE_TIMER_BUFFER_COUNT; i++) {
+        printf("buffer[%03d] = %lu\r\n", i, uvisor_ctx->buffer[i]);
+    }
+}
+
+/* Secure box main thread function */
+static void private_timer_main_thread(const void *)
+{
+    /* Create the buffer and cache its pointer to the private static memory. */
+    uvisor_ctx->buffer = (uint32_t *) malloc(PRIVATE_TIMER_BUFFER_COUNT * sizeof(uint32_t));
+    if (uvisor_ctx->buffer == NULL) {
+        mbed_die();
+    }
+    uvisor_ctx->index = 0;
+
+    /* Setup the push-button callback. */
+    InterruptIn button(SW2);
+    button.mode(PullUp);
+    button.fall(&private_timer_button_on_press);
+
+    /* Setup and start the timer. */
+    Timer timer;
+    timer.start();
+
+    while (1) {
+        /* Store the timer value. */
+        uvisor_ctx->buffer[uvisor_ctx->index] = timer.read_us();
+
+        /* Update the index. Behave as a circular buffer. */
+        if (uvisor_ctx->index < PRIVATE_TIMER_BUFFER_COUNT - 1) {
+            uvisor_ctx->index++;
+        } else {
+            uvisor_ctx->index = 0;
+        }
+    }
+}
+```
+
+A few things to note in the code above:
+
+* If code is running in the context of `private_timer`, then any object instantiated inside that code will belong to the `private_timer` heap and stack. This means that in the example above the `InterruptIn` and `Timer` objects are private to the `private_timer` box. The same applies to the dynamically allocated buffer `uvisor_ctx->buffer`.
+* The content of the private memory `PrivateTimerStaticMemory` can be accessed using the `PrivateTimerStaticMemory * uvisor_ctx` pointer, which is maintained by uVisor.
+* The `InterruptIn` object triggers the registration of an interrupt slot. Since that code is run in the context of the `private_timer` box, then the push-button IRQ belongs to that box. If you want to use the IRQ APIs directly, read the [section](#the-nvic-apis) below.
+* Even if the `private_timer_button_on_press` function runs in the context of `private_timer`, we can still use the `printf` function, which accesses the `UART0` peripheral, owned by the main box. This is because all ACLs declared in the main box are by default shared with all the other secure boxes. This also means that the messages we are printing on the serial port are not secure, because other boxes have access to that peripheral.
+
+> **Note**: Instantiating an object in the `secure_box.cpp` global scope will automatically map it to the main box context (heap), not the `private_timer` one. If you want an object to be private to a box, you need to instantiate it inside the code that will run in the context of that box.
```

Comment:
Maybe this should be stronger than a `Note`. Perhaps a `Caution` or `Look out!` or whatever is standard in mbed documentation for `You are about to make a mistake!`.